pkgs/os-specific/linux/kernel/manual-config.nix

```diff
@@ -417,7 +425,9 @@ stdenv.mkDerivation ((drvAttrs config stdenv.hostPlatform.linux-kernel kernelPat
 
   enableParallelBuilding = true;
 
-  hardeningDisable = [ "bindnow" "format" "fortify" "stackprotector" "pic" "pie" ];
+  hardeningDisable = [ "bindnow" "format" "fortify" "stackprotector" "pic" "pie" ]
+    # Clang does not support zerocallusedregs on arm
```

Comment:
Should we encode this in the cc-wrapper instead?

Reply:
Probably, it's only with thumb I saw so we'd have to change the check a little bit.

Reply:
It looks like this shouldn't be coded into the cc-wrapper, `hardeningUnsupportedFlagsByTargetPlatform` already adds `zerocallusedregs` when the target platform is a 32-bit ARM system. However, the way Linux invokes stuff we cannot dynamically switch out the compiler that way. Unless we could figure out how to have cc-wrapper identify hardening options which aren't supported when changing between 64 and 32 bit.

Reply:
I'd expect the build system to pass `-m32` or similar in that case.

Reply:
Yeah, should we try a build and see what flags it passed so we can code cc-wrapper to drop the unsupported hardening flag in that case?

Reply:
Got this
```
# CC32    arch/arm64/kernel/vdso32/note.o
  /nix/store/hln7mn61cjzsnqwy0p392kgsiq3k5c2v-aarch64-unknown-linux-gnu-clang-wrapper-18.1.8/bin/aarch64-unknown-linux-gnu-cc --target=arm-linux-gnueabi -Wp,-MD,arch/arm64/kernel/vdso32/.note.o.d -DBUILD_VDSO -D__KERNEL__ -nostdinc -isystem /nix/store/hln7mn61cjzsnqwy0p392kgsiq3k5c2v-aarch64-unknown-linux-gnu-clang-wrapper-18.1.8/resource-root/include -I./arch/arm64/include -I./arch/arm64/include/generated  -I./include -I./arch/arm64/include/uapi -I./arch/arm64/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/compiler-version.h -include ./include/linux/kconfig.h -g -mabi=aapcs-linux -mfloat-abi=soft -mlittle-endian -fPIC -fno-builtin -fno-stack-protector -DDISABLE_BRANCH_PROFILING -march=armv8-a -DENABLE_COMPAT_VDSO=1 -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -Werror-implicit-function-declaration -Wno-format-security -std=gnu11 -O2 -fno-strict-overflow -Werror=date-time -D__uint128_t='void*' -Wno-int-to-pointer-cast -mthumb -fomit-frame-pointer -c -o arch/arm64/kernel/vdso32/note.o arch/arm64/kernel/vdso32/note.c
```

Reply:
#335023

Reply:
Ditched the ARM fix for the changes introduced in the PR linked in my previous comment.